calliope/postprocess/results.py

```diff
@@ -91,10 +91,15 @@ def capacity_factor(results, model_data, systemwide=False):
 
     """
     # In operate mode, energy_cap is an input parameter
-    if "energy_cap" not in results.keys():
-        energy_cap = model_data.energy_cap
-    else:
+    try:
         energy_cap = results.energy_cap
+    except AttributeError:
+        try:
+            energy_cap = model_data.energy_cap
+        except AttributeError:
```

Comment:
Is this a smart idea?

Reply:
The relative benefit of if/else vs try/except seems to be hotly debated on forums; I would be happy enough using if/else if you think it's smarter. TBH, our operate mode stuff *isn't* very smart, as evidenced by things like this.